openlibrary/macros/ReadingLogButton.html

```diff
@@ -2,18 +2,18 @@
 
 <form method="POST" action="$(work.key)/bookshelves.json?debug=true" class="reading-log-lite">
   <select name="bookshelf_id">
-    $if read_status == 3:
-      <option value="3"><span class="activated-check">✓</span> Already Read</option>
-      <option value="1">Want to Read</option>
-      <option value="2">Currently Read</option>
-    $elif read_status == 2:
-      <option value="2"><span class="activated-check">✓</span> Currently Reading</option>
-      <option value="1">Want to Read</option>
-      <option value="3">Already Read</option>
-    $else:
-      <option value="1"><span class="activated-check">✓</span> Want to Read</option>
-      <option value="2">Currently Reading</option>
-      <option value="3">Already Read</option>
-    <option value="$(read_status)">Remove</option>
+    $for value in [1,2,3]:
```

Comment:
Let's not reinvent a `dict` using `if/else` (again 😅 )

```suggestion
    $ shelves_by_id = {1: _('Want to Read'), 2: _('Currently Reading'), 3: _('Already Read'), -1: _('Remove') }
    $for shelf_id in [1, 2, 3, -1]:
        $ is_selected = read_status == shelf_id
        <option value="$shelf_id" $cond(is_selected, 'selected', '')>
        $ if is_selected:
            <span class="activated-check">✓</span>
        $shelves_by_id[shelf_id]
```

Reply:
Done